viewing_party/party.py

```diff
@@ -0,0 +1,208 @@
+            return None
+        for movie_name in values:
+          
+            if movie_name["genre"] not in max_genre:
+               
+               max_genre[movie_name["genre"]]=1
+               
+            else:
+           
+              max_genre[movie_name["genre"]]+=1
+              
+
+           
+    max_num=max_genre.values()
+    max_num=max(max_num)
+
+
+    for key, value in max_genre.items():
+        if value == max_num:
+            return key
+
+
+# test_wave_03
+
+def get_unique_watched(amandas_data):
+    friends_watched_list=[]
+    result=[]
+    for friend in amandas_data["friends"]:
+        for movie in friend["watched"]:
+             friends_watched_list.append(movie["title"])
+             print("ss",movie["title"])
+                   
+    for watched_list in amandas_data["watched"]:
+        for key_val,vals in watched_list.items():
+            if vals not in friends_watched_list:
+                print("ddd",vals)
+                result.append(watched_list)
+    return result
+
+
+def get_friends_unique_watched(amandas_data):
+    user_watched_list=[]
+    result=[]
+    for watched in amandas_data["watched"]:
+        print(watched)
+      
+        user_watched_list.append(watched)
+    for friends_watched in amandas_data["friends"]:
+        for watched, watched_list in friends_watched.items():
+            print("ss",watched_list)
+            for val in watched_list:
+                print("aas",val)
+                if val not in user_watched_list:
+                    result.append(val)
+    return [dict(t) for t in {tuple(d.items()) for d in result}]
+
+
+# test_wave_04
+
+
+def get_available_recs(amandas_data):
+    print("hh",amandas_data["watched"])
+    empty_user_watched_list=True
+    friend_watched_list=[]
+    
+    result=[]
+    for watched in amandas_data["friends"]:
+      
+        for key_val,val_list in watched.items():
+       
+            for val in val_list:
+               
+                friend_watched_list.append(val)
+    print(friend_watched_list)
+
+    while empty_user_watched_list:
+        for item in friend_watched_list:
+            
+            if item["host"] in amandas_data["subscriptions"]:
+                result.append(item)
+        
+        empty_user_watched_list=False
+
+    return [dict(t) for t in {tuple(d.items()) for d in result}]
+
+
+# test_wave_05
+def get_new_rec_by_genre(sonyas_data):
+    genre=[]
+    user_watched=[]
+    result=[]
+    for user_movie in sonyas_data["watched"]:
+            if user_movie["genre"] not in genre:
+               genre.append(user_movie["genre"])
+            user_watched.append(user_movie["title"])
+            print("dd",genre,user_watched)
+    
+    for friends_movie in sonyas_data["friends"]:
+        for movie in friends_movie["watched"]:
+           if movie["title"] not in user_watched and movie["genre"] in genre:
+              print("hhh")
+              result.append(movie)
```

Comment:
This could end up recommending a movie that's not in the user's favorite genre as long as it's one they haven't otherwise seen. Could we start with a function that gets the friend movies the user definitely hasn't seen, and also used a function that gets the user's favorite genre. How could we combine those results to find a solution to this feature?